BasicGates/BasicGates.ipynb

```diff
@@ -546,7 +546,52 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Task 2.4. Toffoli gate\n",
+    "### Task 2.4. Two-qubit gate - 4\n",
+    "\n",
+    "**Input:** Two unentangled qubits (stored in an array of length 2) in an arbitrary two-qubit state ${\\color{blue}\\alpha} |00\\rangle + {\\color{blue}\\beta} |01\\rangle + \\gamma |10\\rangle + \\delta |11\\rangle$.\n",
+    "\n",
+    "\n",
+    "**Goal:**  Change the two-qubit state to ${\\color{red}\\beta} |00\\rangle + {\\color{red}\\alpha} |01\\rangle + \\gamma |10\\rangle + \\delta |11\\rangle$.\n",
+    "\n",
+    "> This task can be solved using one library function; as an exercise, try to express the solution using several (possibly controlled) Pauli gates.  \n",
+    ">\n",
+    "> If you can't come up with a solution, or if you have solved the task, read about the relevant library function: \n",
+    "  <details> \n",
+    "    <summary><b>Click here.</b></summary>\n",
+    "    `ControlledOnInt` function can generate a controlled gate, which is applied only when the control qubit is in the $|0\\rangle$ state. </br>\n",
+    "    For example, `ControlledOnInt(0, H)([qs[0]], qs[1])` applies the Hadamard gate to the `qs[1]` qubit, when the control qubit `qs[0]` is in the $|0\\rangle$ state.\n",
+    "    </br></br>\n",
+    "    Notice that the `ControlledOnInt` funciton requires the control qubit(s) to be passed as a register - this function generates a gate controlled by multiple qubits. The gate is applied if the register is in a state equivalent to a non-negative integer (0 in the example above). A similar function is `ControlledOnBitString` which will be introudced in [Multi-Qubit Gates tutorial](../tutorials/MultiQubitGates/MultiQubitGates.ipynb).\n",
```

Comment:
Possibly some typos here and a bit of rewording
```suggestion
    "    Notice that the `ControlledOnInt` function requires the control qubit(s) to be passed as a register. This function generates a gate controlled by multiple qubits. The gate is applied if the control register state corresponds to the specified bit mask (0 in the first example above). A similar function is `ControlledOnBitString` which will be introduced in [Multi-Qubit Gates tutorial](../tutorials/MultiQubitGates/MultiQubitGates.ipynb).\n",
```

Reply:
About the rewording and typos: I mostly agree, although I think we should be consistent with using "a non-negative integer" instead of "the specified bit mask"

Reply:
I like "the specified bit mask" instead of "non-negative integer" as it specifies things more clearly. Only place I can see this in "SuperPosition" kata in hint added by you. So, you might want to rename it in this PR to be more consistent

Reply:
I try to avoid using the "specified bit mask" to make a distinction between `ControlledOnInt` and `ControlledOnBitString`. I've added an additional sentence explaining that the "non-negative integer" specifies a bit mask. What do you think about such explanation?

Reply:
This way we are consistent with [documentation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.canon.controlledonint).
Now there is a "positive integer" specified there, but I've already created an issue to change that to a "non-negative integer".

Reply:
About navigating through the notebooks: 
When I was going through the "Quantum Computing Concepts: Qubits and Gates" section, I used the order from the "Learning path" as the order-in-which-notebooks-should-be-completed. It's probably a personal trait but I treat conclusion links as links-to-notebooks-that-you-may-want-to-have-a-look-at, even though it's stated that they are the next-learning-step.

It would be interesting to hear from @tcNickolas about this.

Reply:
Regarding https://github.com/microsoft/QuantumKatas/pull/827#discussion_r949419262, sounds good.

Reply:
Catching up with this thread... Even before this change, the second part of the BasicGates kata listed MultiQubitGates tutorial as a resource. I think in this case we have either a loop of references or too many prerequisites for the BasicGates, since the kata covers both single- and multi-qubit basic gates, and the tutorials for them are split in two. We can clarify the final reference in the MultiQubitGates tutorial, saying "You should start with _the second part of_ the BasicGates kata", but other than that, I think it's fine as is.

Do you want to send a PR to add that clarification to the MultiQubitGates tutorial conclusion?